models/Task.php

```diff
@@ -1143,4 +1153,70 @@ private function getWorker()
     {
         return User::findOne(['id' => $this->getAccount(Task::ACCOUNT_WORKER)->user_id])->guid;
     }
+
+    /**
+     * revert all transactions where task space account is the target account
+     */
+    public function refund()
+    {
+        $incomeTransactions = Transaction::findAll(['to_account_id' => $this->getAccount(Task::ACCOUNT_SPACE)->id]);
+
+        foreach ($incomeTransactions as $transaction) {
+            $transaction->delete();
+        }
+    }
+
+
+    /**
+     * Transfer coins from task space account to worker target account
+     */
+    public function payWorker()
+    {
+        $fromAccount = $this->getAccount(Task::ACCOUNT_SPACE);
+        $toAccount = $this->getAccount(Task::ACCOUNT_WORKER);
+
+        foreach ($fromAccount->getAssets() as $asset) {
+            $incomeTransactions = Transaction::findAll([
+                'asset_id' => $asset->id,
+                'to_account_id' => $fromAccount->id
+            ]);
+
+            foreach ($incomeTransactions as $transaction) {
+                $outcomeTransaction = new Transaction();
+                $outcomeTransaction->transaction_type = Transaction::TRANSACTION_TYPE_TASK_PAYMENT;
+                $outcomeTransaction->from_account_id = $fromAccount->id;
+                $outcomeTransaction->to_account_id = $toAccount->id;
+                $outcomeTransaction->asset_id = $asset->id;
+                $outcomeTransaction->amount = $transaction->amount;
+                $outcomeTransaction->comment = "Payment for <<{$this->title}>> task";
+
+                $outcomeTransaction->save();
+            }
```

Comment:
shouldn't we only do one transaction with sum of the input transactions as amount?

if a user completes the task, then one transfer of the coins is made the the receiving account.

Reply:
We can't make a one transaction because for the moment a task can be funded by multiple assets so for each asset we must have a separate transaction. I left it like this to ensure tracability so that worker can see from where the amounts came from. But yes it's better to make one transaction per asset but we can't make only one transaction for all assets. 
What do you think ?

Reply:
that would then be one transaction per asset? ok.

Reply:
Do you agree on one transaction per asset ? Should i make the update ?